tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -461,6 +461,36 @@ public function test_next_tag_should_stop_on_closers_only_when_requested() {
 		$this->assertTrue( $p->is_tag_closer(), 'Indicated a tag closer is a tag opener' );
 	}
 
+	/**
+	 * @ticket 57852
+	 *
+	 * @dataProvider data_rcdata_and_script_tags
+	 * @covers WP_HTML_Tag_Processor::next_tag
+	 * @covers WP_HTML_Tag_Processor::is_tag_closer
+	 *
+	 * @param string $tag_name The name of the tag to test.
+	 */
+	public function test_next_tag_should_stop_on_rcdata_and_script_tag_closers_when_requested( $tag_name ) {
+		$p = new WP_HTML_Tag_Processor( "<$tag_name>abc</$tag_name>" );
```

Comment:
What about when there's a mismatch between the opening and closing tag? ie the unhappy path

Reply:
this is already covered by a number of other tests. I think the addition here is specific to supporting _stopping_ on those tag closers when present.

Reply:
I'm not seeing a unhappy path test that is seeking to stop on the tag closers when it doesn't exist. What am I missing?

I searched for:

```php
array( 'tag_closers' => 'visit' )
```

Reply:
aha, well, I don't see this test in the general case, which actually highlights a comment I almost left, which is that I don't know if these tags warrant a separate test than `next_tag_stops_on_tag`

we have `test_next_tag_should_stop_on_closers_only_when_requested` which covers querying tag closers and `test_next_tag_should_return_false_for_a_non_existing_tag` which covers querying for things that don't exist.

maybe we add these cases into those tests, which focus on those specific tag closer behaviors?

either way it's fine. like I said, I almost suggested we avoid special-casing these here in these tests, but I refrained because I didn't think it was important enough. special-casing them in the tag closer query is probably similarly benign, even if marginally awkward

Reply:
>maybe we add these cases into those tests, which focus on those specific tag closer behaviors?

I'm fine with that idea as long as an `@ticket` annotation is added.

>I almost suggested we avoid special-casing these here in these tests, but I refrained because I didn't think it was important enough. 

What happens though when passing `array( 'tag_closers' => 'visit' )` when the tag closer doesn't exist? By having a specific test to check, it's documenting the expected behavior while covering this use case. Let's imagine in the wild a plugin passes a variable of HTML into `WP_HTML_Tag_Processor( $html );` and doesn't know if there's a closing tag or not. So they follow what the test is doing:

```php
$p = WP_HTML_Tag_Processor( $html );
$p->next_tag();
$closer_tag = $p->next_tag( array( 'tag_closers' => 'visit' ) );
```

What happens when none is found?

My point is: There's currently no test in the suite that is using this code pattern `$p->next_tag( array( 'tag_closers' => 'visit' ) );` to check for when there's no closing tag.  IMO it is worth adding a test to document what is expected and guard the code from future regressions.

Reply:
> By having a specific test to check, it's documenting the expected behavior while covering this use case.

Right, this is where I was agreeing with you 😉 

> There's currently no test in the suite that is using this code pattern

Yes, and I think the reason was that we didn't see this as a separate behavior, given that the case of "searching for a matching tag that doesn't exist" _is_ tested, whereby the tag closer is no different than tag openers.

I agree that adding this is good, just noting that maybe it belongs in its own test instead of being tied in here, but either way, I really don't care.

Reply:
>I agree that adding this is good, just noting that maybe it belongs in its own test instead of being tied in here
I agree with you @dmsnell! Thank you :)

Reply:
Solved in https://github.com/WordPress/wordpress-develop/pull/4164/commits/6e35acb84084df863b517e676e8785958099a331